p2panda-rs/src/secret_group/mls/provider.rs

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+use memory_keystore::MemoryKeyStore;
+use openmls_traits::OpenMlsCryptoProvider;
+use rust_crypto::RustCrypto;
+
+/// Implements the `OpenMlsCryptoProvider` trait to be used as a crypto and key store backend for
+/// all other MLS structs and methods.
+///
+/// @TODO: This will use our own key store soon.
+#[derive(Debug)]
+pub struct MlsProvider {
```

Comment:
Of all the elements in this module I feel the provider is the hardest to understand conceptually, I wonder if the docs need to be expanded a little more to explain it's function, maybe even where it sits in the MLS architecture and the way we're implementing it?

Reply:
Yeah, this part is a bit abstract, I've tried to clarify it a little more in a comment, does this make more sense now?